SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -465,6 +469,12 @@ - (void)enqueueAttachment:(DataSource *)dataSource
             SignalRecipient *newRecipient = [self.contactsUpdater synchronousLookup:recipientId error:error];
             if (newRecipient) {
                 [recipients addObject:newRecipient];
+            } else {
+                DDLogWarn(@"%@ No SignalRecipient for recipientId: %@", self.logTag, recipientId);
+                [self.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
+                    // Mark this recipient as "skipped".
+                    [message updateWithSkippedRecipient:recipientId transaction:transaction];
```

Comment:
Mark unregistered recipients as skipped.

Reply:
Since the calling method is already responsible for doing async/dispatch and creating transactions, can we pass this transaction in rather than create one here?

It seems like it would be easy to break it, creating a nested transaction if the caller gets refactored, and not notice since this is not happy path.

Reply:
Yup. I'm trying to use a narrow, short-lived transaction here since:

* It's an edge case.
* I don't want to bake in the assumption that `[contactsUpdater synchronousLookup]` doesn't use a transaction.

As I mentioned I'd like to refactor this sending logic soon for many reasons, and you're pointing to additional worthy goals for that refactor.  I've updated the card.